src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -473,67 +473,48 @@ private BigInteger(int n, uint[]? rgu)
 
         /// <summary>
         /// Constructor used during bit manipulation and arithmetic.
-        /// When possible the span will be packed into _sign to conserve space.
+        /// When possible the value will be packed into  _sign to conserve space.
         /// </summary>
         /// <param name="value">The absolute value of the number</param>
-        /// <param name="valueArray">An array, identical in contents to <paramref name="value"/>, for which ownership can be transferred; null if no such array was available.</param>
         /// <param name="negative">The bool indicating the sign of the value.</param>
-        private BigInteger(ReadOnlySpan<uint> value, uint[]? valueArray, bool negative)
+        private BigInteger(ReadOnlySpan<uint> value, bool negative)
         {
-            Debug.Assert(valueArray is null || value.SequenceEqual(valueArray));
+            int len;
 
-            // Try to conserve space as much as possible by checking for wasted leading uint entries
-            // (sometimes the span has leading zeros from bit manipulation operations & and ^).
-            if (!value.IsEmpty && value[^1] == 0)
-            {
-                int len = value.Length - 1;
-                while (len > 0 && value[len - 1] == 0)
-                {
-                    len--;
-                }
-                value = value.Slice(0, len);
-                valueArray = null;
-            }
+            // Try to conserve space as much as possible by checking for wasted leading span entries
+            // sometimes the span has leading zeros from bit manipulation operations & and ^
+            for (len = value.Length; len > 0 && value[len - 1] == 0; len--);
 
-            if (value.IsEmpty)
-            {
+            if (len == 0)
                 this = s_bnZeroInt;
-            }
-            else if (value.Length == 1 && value[0] < kuMaskHighBit)
+            // Values like (Int32.MaxValue+1) are stored as "0x80000000" and as such cannot be packed into _sign
```

Comment:
This comment should remain attached to the `_sign` assignment

Reply:
Fixed